inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.plugin.task;
+
+import org.apache.inlong.agent.conf.InstanceProfile;
+import org.apache.inlong.agent.conf.TaskProfile;
+import org.apache.inlong.agent.constant.TaskConstants;
+import org.apache.inlong.agent.core.instance.ActionType;
+import org.apache.inlong.agent.core.instance.InstanceAction;
+import org.apache.inlong.agent.core.instance.InstanceManager;
+import org.apache.inlong.agent.core.task.TaskManager;
+import org.apache.inlong.agent.db.Db;
+import org.apache.inlong.agent.metrics.audit.AuditUtils;
+import org.apache.inlong.agent.plugin.file.Task;
+import org.apache.inlong.agent.state.State;
+import org.apache.inlong.agent.utils.AgentUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+public abstract class AbstractTask extends Task {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractTask.class);
+    public static final int CORE_THREAD_SLEEP_TIME = 1000;
+    public static final int CORE_THREAD_PRINT_TIME = 10000;
+    protected TaskProfile taskProfile;
+    protected Db basicDb;
+    protected TaskManager taskManager;
+    private InstanceManager instanceManager;
+    protected volatile boolean running = false;
+    protected boolean initOK = false;
+    protected long lastPrintTime = 0;
+
+    @Override
+    public void init(Object srcManager, TaskProfile taskProfile, Db basicDb) throws IOException {
+        taskManager = (TaskManager) srcManager;
+        this.taskProfile = taskProfile;
+        this.basicDb = basicDb;
+        instanceManager = new InstanceManager(taskProfile.getTaskId(), taskProfile.getInt(TaskConstants.FILE_MAX_NUM),
+                basicDb, taskManager.getTaskDb());
+        try {
+            instanceManager.start();
+        } catch (Exception e) {
+            LOGGER.error("start instance manager error: ", e);
+        }
+        initTask();
+        initOK = true;
+    }
+
+    protected abstract void initTask();
+
+    protected abstract void releaseTask();
+
+    @Override
+    public void destroy() {
+        doChangeState(State.SUCCEEDED);
+        if (instanceManager != null) {
+            instanceManager.stop();
+        }
+        releaseTask();
+    }
+
+    @Override
+    public TaskProfile getProfile() {
+        return taskProfile;
+    }
+
+    @Override
+    public String getTaskId() {
+        if (taskProfile == null) {
+            return "";
+        }
+        return taskProfile.getTaskId();
+    }
+
+    @Override
+    public void addCallbacks() {
+
+    }
+
+    @Override
+    public void run() {
+        Thread.currentThread().setName("task-core-" + getTaskId());
```

Comment:
Better to distinguish the thread name for different types of tasks.

Reply:
At first, I thought the same way, but in reality, different data sources will not be enabled simultaneously in the same process